overlord/state/change.go

```diff
@@ -567,8 +567,13 @@ NextChangeTask:
 		t := c.state.tasks[tid]
 
 		var live bool
-		switch t.Status() {
-		case DoStatus, DoingStatus, WaitStatus, DoneStatus:
+		status := t.Status()
+		if status == WaitStatus {
+			// If the task is waiting, then use the effective status instead.
+			status = t.WaitedStatus()
```

Comment:
As this pattern is repeated a couple of times, maybe it would make sense to have a method in the task to abstract this (`task.EffectiveStatus()` or similar)

Reply:
Done! As this behaviour is limited to Change.go I have added a helper method instead of adding a member method. I don't want to expose to many `*Status` functions on the Task structure as this would be confusing.